### DISCUSSION TITLE
Could I create the defination of a method declared in the header file in a quick way ? （in vscode）

### DISCUSSION THREAD

**Discussion by longfar-ncy** [clangd/clangd#1935] [Q&A]
Could I create the defination of a method declared in the header file in a quick way ? It's like the C++ extension plugin. For methods that have no definition but only declarations, a single button generates the definition in a cpp file with the same name as the header file

**Comment by longfar-ncy** [comment]
Here's a video to show the feature in C++ extension:
[video_link](https://vdn3.vzuu.com/SD/9218603c-a4a2-11ed-bbd9-b67290458ebb-v1_f4_t2_lyJ2kL3N.mp4?disable_local_cache=1&bu=078babd7&c=avc.1.1&f=mp4&expiration=1708017297&auth_key=1708017297-0-0-25213b969bd10c543990bea0910bb874&v=tx&pu=078babd7)

**Comment by HighCommander4** [comment]
The current way is:
 * change the declaration to an empty definition (replace `;` with `{}`)
 * perform the "define out of line" code action at the empty definition

See also https://github.com/clangd/clangd/issues/1776

**Reply by longfar-ncy** [reply]
thx very much
